packages/sync/src/NetworkStatus.ts

```diff
@@ -0,0 +1,25 @@
+class NetworkStatus {
+
+  /**
+   *
+   * Trigger a function whenever the user switches into "Online Mode"
+   *
+   * @param fn Function to be called when got online
+   */
+  public static whenOnline(fn: any): void {
+    window.addEventListener("online", fn, false);
+  }
+
+  /**
+   *
+   * Trigger a function whenever the user switches into "Offline Mode"
+   *
+   * @param fn Function to be called when got offline
+   */
+  public static whenOffline(fn: any): void {
+    window.addEventListener("offline", fn, false);
+  }
+
+}
+
+export default NetworkStatus;
```

Comment:
Avoid default exports. 
This is not going to work with javascript and it will require `const interface = require("x").networkInterface.default`